src/Common/src/System/Net/HttpKnownHeaderNames.TryGetHeaderName.cs

```diff
@@ -47,6 +50,36 @@ public static unsafe bool TryGetHeaderName(IntPtr buffer, int length, out string
                 out name);
         }
 
+        public static string GetHeaderValue(string name, char[] array, int startIndex, int length)
+        {
+            Debug.Assert(name != null);
+            CharArrayHelpers.DebugAssertArrayInputs(array, startIndex, length);
+
+            if (length == 0)
+            {
+                return string.Empty;
+            }
+
+            // If it's a known header value, use the known value instead of allocating a new string.
+
+            // Do a really quick reference equals check to see if name is the same object as
+            // HttpKnownHeaderNames.ContentEncoding, in which case the value is very likely to
+            // be either "gzip" or "deflate".
+            if (ReferenceEquals(name, ContentEncoding))
+            {
+                if (CharArrayHelpers.EqualsOrdinalAsciiIgnoreCase(Gzip, array, startIndex, length))
+                {
+                    return Gzip;
```

Comment:
I don't have a strong opinion, really.  It's certainly simpler internally to just normalize these, then we can do ref comparisons in a bunch of cases.  

I'd like to implement some optimizations to header handling in HttpClient proper, and being able to just always normalize the case of header names would be nice.

I'm trying to think of a case where the capitalization on the wire would be interesting to the user, and I'm failing.  Still, it might be a bit unexpected.

It looks like you stole this code from the WinHttpHandler, right?  Meaning we are already doing this normalization for WinHttpHandler?